src/app/pages/account/account.component.ts

```diff
@@ -111,13 +111,28 @@ export class AccountComponent implements OnDestroy {
         void this._router.navigate([`/${hashNavItem.route}/${hash}`], { replaceUrl: true });
     }
 
+    /** Call this method whenever someone has enters a BNS domain, and it is resolved to a Banano address. */
+    private _redirectToAccountPage(account: string): void {
+        void this._router.navigate([`/${accountNavItem.route}/${account}`], { replaceUrl: true });
+    }
+
     /** Given a ban address, searches for account. */
-    private _searchAccount(address): void {
+    private async _searchAccount(address): Promise<void> {
         if (!address) {
             return;
         }
 
         if (!address.startsWith('ban_')) {
+            const parts = address.split('.');
+            if (parts.length === 2) {
+                //search in api
+                try {
+                    const domain = await this.apiService.fetchBNSDomain(parts[0], parts[1]);
+                    if (domain.domain?.resolved_address) {
+                        return this._redirectToAccountPage(domain.domain?.resolved_address);
+                    }
+                } catch (_) {}
+            }
```

Comment:
Could we get a quick comment here explaining what this does?  If not a ban address, assume it's BNS?